src/Core/Managed/Shared/Extensibility/Implementation/BroadcastProcessor.cs

```diff
@@ -0,0 +1,148 @@
+﻿namespace Microsoft.ApplicationInsights.Shared.Extensibility.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    /// <summary>
+    /// An <see cref="ITelemetryProcessor"/> that asynchronously sends the data to multiple child telemetry processors.
+    /// </summary>
+    internal class BroadcastProcessor : ITelemetryProcessor, IDisposable
```

Comment:
OK, so I think you are saying we should execute sink processors synchronously and rely on the fact that both processors and `channel.Send` are implemented efficiently (with channel having its own internal queueing). With these assumptions the synchronous "simple iteration" variant will have acceptable performance.

I will put a comment to that effect int he `BroadcastProcessor` code and make it synchronous. I will also probably remove the AsyncCall implementation from the PR since it won't be used by the rest of the code anymore. It is part of the GitHub history now and we can resurrect it should it become necessary for some future channel implementation.